test/unit/spec/LoginRouter_spec.js

```diff
@@ -170,48 +170,75 @@ function (Okta, Q, Backbone, SharedUtil, CryptoUtil, Logger, OktaAuth, Util, Exp
       });
     }
 
+    function expectPrimaryAuthRender(options = {}, path = '') {
+      // Reusable stub to assert that the Primary Auth for renders
+      // given Widget parameters and a navigation path.
+      return setup(options)
+      .then(function (test) {
+        Util.mockRouterNavigate(test.router);
+        test.router.navigate(path);
+        return Expect.waitForPrimaryAuth();
+      })
+      .then(function () {
+        var form = new PrimaryAuthForm($sandbox);
+        expect(form.isPrimaryAuth()).toBe(true);
+      });
+    }
+
     function expectUnexpectedFieldLog(arg1) {
       // These console warnings are called from Courage's Logger class, not
       // the Widget's. We need to assert that the following is called in specific
       // environments (window.okta && window.okta.debug are defined).
-      expect(Logger.warn).toHaveBeenCalledWith('Field not defined in schema', arg1);
+      expect(CourageLogger.warn).toHaveBeenCalledWith('Field not defined in schema', arg1);
     }
 
     it('logs a ConfigError error if unknown option is passed as a widget param', function () {
-      spyOn(Logger, 'warn');
+      spyOn(CourageLogger, 'warn');
       var fn = function () { setup({ foo: 'bla' }); };
-      expect(fn).not.toThrow(Errors.ConfigError);
+      expect(fn).not.toThrowError(Errors.ConfigError);
       expectUnexpectedFieldLog('foo');
     });
     it('has the correct error message if unknown option is passed as a widget param', function () {
-      spyOn(Logger, 'warn');
+      spyOn(CourageLogger, 'warn');
       var fn = function () { setup({ foo: 'bla' }); };
       expect(fn).not.toThrow();
       expectUnexpectedFieldLog('foo');
     });
-    it('throws a ConfigError if el is not passed as a widget param', function () {
+    it('logs a ConfigError error if el is not passed as a widget param', function () {
       var fn = function () { setup({ el: undefined }); };
-      expect(fn).toThrowError(Errors.ConfigError);
+      expect(fn).not.toThrow();
+      expect(Logger.warn).toHaveBeenCalled();
     });
     it('has the correct error message if el is not passed as a widget param', function () {
       var fn = function () { setup({ el: undefined }); };
-      expect(fn).toThrowError('"el" is a required widget parameter');
+      expect(fn).not.toThrow();
+      var err = Logger.warn.calls.mostRecent().args[0];
+      expect(err.name).toBe('CONFIG_ERROR');
+      expect(err.message).toEqual('"el" is a required widget parameter');
     });
-    it('throws a ConfigError if baseUrl is not passed as a widget param', function () {
+    it('logs a ConfigError if baseUrl is not passed as a widget param', function () {
```

Comment:
@jmelberg-okta What ends up being sent to the console here, a warning?

Reply:
This is a misleading test - as `okta-auth-js` throws an error before the Widget has a chance to evaluate this.

I've updated this PR to expect an `AuthSdkError`, instead of asserting a `console` warning that will **never** be seen.